src/components/AggregationSettings/OpticsReducer.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react'
+import { Box, Button, Text } from 'grommet'
+import { Field, Formik } from 'formik'
+import styled from 'styled-components'
+import { func } from 'prop-types'
+import * as Yup from 'yup'
+import { REDUCERS } from './AggregationSettingsContainer'
+
+const opticsSchema = Yup.object().shape({
+  auto: Yup.boolean(),
+  minSamples: Yup.number()
+    .min(2, 'Must be at least two')
+    .max(50, 'Too long'),
+  xi: Yup.number()
+    .min(0, 'Must be no less than 0')
+    .max(1, 'Must not be greater than 1')
+    .required('Required'),
+  angleEps: Yup.number()
+    .min(0, 'Must be at least 0')
+    .max(180, 'Must not be greater than 180')
+    .required('Required'),
+  gutterEps: Yup.number()
+    .min(0, 'Must be at least 0')
+    .required('Required'),
+  minLineLength: Yup.number()
+    .min(0, 'Must be at least 0')
+    .required('Required'),
+});
+
+const StyledNumberInput = styled(Field)`
+  border: 1px solid #979797;
+  border-radius: 0.25em;
+  height: 2em;
+  text-align: center;
+`
+
+const StyledLabel = styled.label`
+  font-size: 0.75em;
+`
+
+export default function OpticsReducer({ setCallback, setScreen, submitOptics }) {
+  return (
+    <Box gap='small' direction='row'>
+      <Formik
+        initialValues={{
+          angleEps: 30,
+          auto: false,
+          gutterEps: 150,
+          minLineLength: 0,
+          minSamples: 2,
+          xi: 0.05,
+        }}
+        isInitialValid
+        validationSchema={opticsSchema}
+      >
+        {({ errors, isValid, values }) => (
+          <Box as='form' onSubmit={submitOptics} gap='small'>
+            <Box gap='xsmall' pad='small'>
+              <Text>OPTICS Reducer</Text>
+
+              <Box direction='row' gap='xsmall'>
+                <Box basis='large'>
+                  <Text weight='bold'>min_samples</Text>
+                  <Text>
+                    Increase if a single line of text is being identified multiple times,
+                    decrease if multiple lines are being clustered together.
+                  </Text>
+                  <Text color='red' margin={{ top: 'xsmall' }} size='xsmall'>{errors && errors.minSamples}</Text>
+                </Box>
+                <Box basis='xsmall'>
+                  <StyledNumberInput name='minSamples' type='number' min='2' />
+                  <Box align='center' direction='row'>
+                    <Field id='auto' name='auto' type='checkbox' checked/>
```

Comment:
Curiously, when testing the "Optics Reducer" form on Storybook, the 'auto' checkbox is the only input field that can't be edited.